tfhe/src/high_level_api/integers/signed/compressed.rs

```diff
@@ -98,3 +106,33 @@ impl<Id: FheIntId> ParameterSetConformant for CompressedFheInt<Id> {
 impl<Id: FheIntId> Named for CompressedFheInt<Id> {
     const NAME: &'static str = "high_level_api::CompressedFheInt";
 }
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub enum CompressedSignedRadixCiphertext {
+    Seeded(IntegerCompressedSignedRadixCiphertext),
+    ModulusSwitched(CompressedModulusSwitchedSignedRadixCiphertext),
+}
+
+impl ParameterSetConformant for CompressedSignedRadixCiphertext {
+    type ParameterSet = RadixCiphertextConformanceParams;
+    fn is_conformant(&self, params: &RadixCiphertextConformanceParams) -> bool {
+        match self {
+            Self::Seeded(ct) => ct.is_conformant(params),
+            Self::ModulusSwitched(ct) => ct.is_conformant(params),
+        }
+    }
+}
+
+impl<Id> FheInt<Id>
+where
+    Id: FheIntId,
+{
+    pub fn compress(self) -> CompressedFheInt<Id> {
```

Comment:
I would take `&self` here

Reply:
it has been discussed, refactor incoming to stop taking things ownership/by value unless the code requires it, especially old code with decompression and such, you were right on that one my bad !

Reply:
Done